lib/ansible/modules/cloud/cloudstack/cs_instance_password_reset.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2018, Gregor Riepl <onitake@gmail.com>
+# based on cs_sshkeypair (c) 2015, René Moser <mail@renemoser.net>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: cs_instance_password_reset
+short_description: Allows resetting VM root passwords on Apache CloudStack based clouds.
+description:
+    - Reset the root password on a virtual machine.
+    - Requires cloud-init installed in the virtual machine.
```

Comment:
I was certain that cloud-init already takes care of that, but let me double-check.
Here: https://github.com/number5/cloud-init/blob/master/cloudinit/sources/DataSourceCloudStack.py#L137-L144
This will cause cloud-init to enable password authentication in sshd_config. Depending on how the defaults of cloud-init are configured, it will reset after the next reboot or be kept enabled.
(I actually tested this recently)

Reply:
It seems Cloudstack first [check if template has `passwordenabled` param set to `true`](https://github.com/apache/cloudstack/blob/master/server/src/main/java/com/cloud/vm/UserVmManagerImpl.java#L672).
According to the [API doc](https://cloudstack.apache.org/api/apidocs-4.11/apis/registerTemplate.html), default is false.

Reply:
I see... sorry I'd overlooked that you referred to the VM template.